### PR TITLE
fix: Fix SeriesKey sort order for special _measurement and _field

### PR DESCRIPTION
**Pull request by mkmik**
Closes https://github.com/influxdata/conductor/issues/1051

I ran this draft PR against the flux test suite we run on the CI:

```
$ (export FLUXTEST_IOX_GRPC_ADDR=localhost:8082; export FLUXTEST_IOX_HTTP_ADDR=localhost:8080; eval $(jsonnet -S -e '(import ".circleci/workflow-nightly-flux.libsonnet").jobs["test-flux-integration"].steps[5].run.command'))
...
---
Found 807 tests: passed 642, failed 0, skipped 165
```

Just to be super sure the tests actually test the code I think it tests, I intentionally broke the code with:

```
                //                key: FIELD_COLUMN_NAME.into(),
                key: "_field_bad".into(),
```

and indeed the test suite complained loudly.

---

TODO
* fix all the IOx tests that depended on the old order

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
The code looks good to me -- thank you @mkmik  🙏 

I am going to verify that the tests pass locally for me (I am sure they will) and will update the remaining tests for this PR and hopefully will get this merged later today (which will hopefully reveal the next big tranche of discrepancies).

**Review by alamb — APPROVED** [review]
I verified that the fluxtests also pass for me locally (thanks for the tip about how to get the command!) 

```
---
Found 807 tests: passed 642, failed 0, skipped 165
(arrow_dev) alamb@MacBook-Pro-8:~/Software/idpe$ 
```


cc @philjb

**Comment by philjb** [comment]
If we've gotten doubly twisted and we need to revert this pr, that's straightforward, yes? I assess that it is based on looking at the code changes but am asking in case of unknown (to me) unknowns.

**Comment by alamb** [comment]
> If we've gotten doubly twisted and we need to revert this pr, that's straightforward, yes? I assess that it is based on looking at the code changes but am asking in case of unknown (to me) unknowns.


@philjb  -- Yes I think so. This code only changes the read path; This PR doesn't change anything about how data is stored (and thus if we need to revert it, a simple `git revert` should do the trick)

**Comment by alamb** [comment]
BTW circle CI is having problems: https://status.circleci.com/ which I think is the source of the `test` job not passing:

Screen Shot 2022-07-19 at 4 48 22 PM

File: service_grpc_influxrpc/src/data.rs [hunk 1]
```diff
@@ -168,7 +169,13 @@ fn convert_tags(tags: Vec<series::Tag>, tag_key_binary_format: bool) -> Vec<Tag>
             },
             value: value.bytes().collect(),
         })
-        .collect()
+        .collect();
+    // tags must be returned in lexicographical order; when we rename the tags
```
Comment (alamb) [inline service_grpc_influxrpc/src/data.rs:173]: 👍

**Pull request merged**